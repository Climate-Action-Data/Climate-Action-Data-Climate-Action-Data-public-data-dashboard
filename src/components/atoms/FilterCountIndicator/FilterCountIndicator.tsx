import { FC } from 'react'
import { Square, Text } from '@chakra-ui/react'

interface FilterCountIndicatorProps {
  count: number
  isResultsPage?: boolean
}

export const FilterCountIndicator: FC<FilterCountIndicatorProps> = (props) => {
  const { count, isResultsPage } = props

  const generateStyle = () => {
    if (isResultsPage) {
      return { backgroundColor: `lightGray.50`, color: `green.900` }
    }
    return { backgroundColor: `green.900`, color: `lightGray.50` }
  }

  return (
    <Square size={`24px`} borderRadius={`4px`} {...generateStyle()}>
      <Text fontFamily={`heading`} fontSize={`12px`}>
        {count}
      </Text>
    </Square>
  )
}
