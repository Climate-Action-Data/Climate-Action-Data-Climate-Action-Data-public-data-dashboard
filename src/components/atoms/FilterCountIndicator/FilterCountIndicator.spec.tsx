import { render } from '@testing-library/react'
import { FilterCountIndicator } from './FilterCountIndicator'

it(`renders correctly`, () => {
  const { container } = render(<FilterCountIndicator count={1} />)
  expect(container).toMatchSnapshot()
})
