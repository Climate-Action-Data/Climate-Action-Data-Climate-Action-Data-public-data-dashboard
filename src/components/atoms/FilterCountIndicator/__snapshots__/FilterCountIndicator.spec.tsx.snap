// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-1vzx6oo"
  >
    <p
      class="chakra-text css-11iqvn7"
    >
      1
    </p>
  </div>
</div>
`;
